File: Data Structures & Algorithms/two-integer-sum/submission-1.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:

        mydict = {}
        mydict[nums[0]] = 0

        for n in range(1,len(nums)):
            if target - nums[n] in mydict:
                return [mydict[target-nums[n]],n]
            if nums[n] not in mydict:
                mydict[nums[n]] = n

        return []
        

File: Data Structures & Algorithms/missing-number/submission-1.cpp
class Solution {
public:
    int missingNumber(vector<int>& nums) {
        int x = 0;
        for (int i = 0; i < nums.size(); i++){
            x ^= nums[i];
        }
        for (int i = 0; i < nums.size()+1; i++){
            x ^= i;
        }
        return x;
    }
};
